### ISSUE TITLE
Aspect ratios can be incorrect

### ISSUE THREAD

**Issue by dankcushions** [libretro/mame2010-libretro#20]
This is the same issue as https://github.com/libretro/mame2003-libretro/issues/51

The fix should be almost identical, but annoyingly the order this core executes is slightly different: here's where the fix should be: https://github.com/libretro/mame2010-libretro/blob/master/src/osd/retro/retromapper.c#L144

but the machine being run isn't initialised till after, here: https://github.com/libretro/mame2010-libretro/blob/master/src/osd/retro/retromain.c#L222
you need the driverindex, etc.

so really either the init needs to happen sooner, or the retro_av_info needs to be reset after the init.

Not sure I want to get stuck into this core so logging the issue.

**Comment by Voljega** [comment]
This issue is still existing today.
For instance, all CPS1 and CPS2 games are displayed with wrong aspect ratios

**Comment by Voljega** [comment]
Removing integer scaling correct the issues on all the games I tested so far

**Comment by dankcushions** [comment]
yes integer scaling will cause problems, but I don't use it. this issue was happening with it turned off (however it may have since been fixed!).
